doc/sets_fragment.txt

```diff
@@ -49,6 +49,34 @@ operation             meaning
 ``excl(A, elem)``     same as ``A = A - {elem}``
 ==================    ========================================================
 
-Sets are often used to define a type for the *flags* of a procedure. This is
-a much cleaner (and type safe) solution than just defining integer
-constants that should be ``or``'ed together.
+Bit fields
+~~~~~~~~~~
+
+Sets are often used to define a type for the *flags* of a procedure or to implement
+bit fields for interoperability with C.
```

Comment:
No, for C interop we nowadays suggest a ``distinct cint`` type, C's enums are just too messy. Also: This should go into the FFI section.

Reply:
The bit layout for Nim sets is incompatible anyway with traditional bit shifting for flags.

Got bitten by this when implementing mmap flags here: https://github.com/numforge/laser/blob/dbfb31d5b25c84d4264377eb6f5b876d6032c766/laser/photon_jit/photon_osalloc.nim#L43-L54

Reply:
The bit layout *can* be compatible but you have to understand how it works. :P

Reply:
Updated.